src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/actions/GenerateTestsUtils.kt

```diff
@@ -62,6 +67,28 @@ fun getSurroundingMethod(psiFile: PsiFile, caret: Caret): PsiMethod? {
     return surroundingMethod
 }
 
+/**
+ * Gets the selected line if the constraints on the selected line are satisfied,
+ *   so that EvoSuite can generate tests for it.
+ * Namely, the line is within a method, it is not blank and contains (part of) a statement.
+ *
+ * @param psiFile the current PSI file (where the user makes the click)
+ * @param caret the current (primary) caret that did the click
+ * @return line number if the constraints are satisfied else null
+ */
+fun getSurroundingLine(psiFile: PsiFile, caret: Caret): Int? {
+    val psiMethod: PsiMethod = getSurroundingMethod(psiFile, caret) ?: return null
+
+    val doc: Document = PsiDocumentManager.getInstance(psiFile.project).getDocument(psiFile) ?: return null
+
+    val selectedLine: Int = doc.getLineNumber(caret.offset)
+    val selectedLineText: String =
+        doc.getText(TextRange(doc.getLineStartOffset(selectedLine), doc.getLineEndOffset(selectedLine)))
+
+    if (selectedLineText.isBlank()) return null
+    return if (!validateLine(selectedLine, psiMethod, psiFile)) null else selectedLine
```

Comment:
Can you maybe move this "if" outside of the return statement to a guard clause in order to improve readability?

Reply:
I thought it would be cool to use the fact that in Kotlin if-statement is an expression. But alright